Vostok.ClusterClient.Transport.Tests/Helpers/SocketTestServer.cs

```diff
@@ -36,9 +37,13 @@ public static SocketTestServer StartNew(string response)
 
         public string LastRequest { get; private set; }
 
-        public void Dispose() => listener.Stop();
+        public void Dispose()
+        {
+            listener.Server.Shutdown(SocketShutdown.Both);
+            listener.Stop();
+        }
 
-        private void Start()
+        private void Start(Action<TcpClient> onBeforeRequestReading)
```

Comment:
It looks like onBeforeRequestReading action is not an argument of the Start method, but part of the SocketTestServer instance. So i think it is better to place it in a class field.

Reply:
fixed